src/xlsx/mod.rs

```diff
@@ -711,6 +719,362 @@ impl<RS: Read + Seek> Xlsx<RS> {
         Ok(())
     }
 
+    #[cfg(feature = "pivot-cache")]
+    // Sets pivot table data. Called by enabling feature pivot-cache.
+    // Unlike tables this is a direct opt-in and only needs to be called once to retrieve supporting information.
+    fn load_pivot_table_metadata(&mut self) -> Result<(), XlsxError> {
+        match self.read_pivot_table_metadata() {
+            Ok(pivot_tables) => {
+                if !pivot_tables.is_empty() {
+                    self.pivot_tables = Some(pivot_tables);
+                }
+                Ok(())
+            }
+            Err(e) => Err(e),
```

Comment:
At a bare minimum we can use `?` here to handle the error path, but this also seems to contradict other functions in that it may lead to `pivot_tables = None` even if they were loaded. Other functions call an unwrap on these `pivot_tables`, like `fn pivot_tables`.